Backend/Routes/users.js

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import bcrypt from 'bcrypt';
+import { User } from '../models/user.js';
+import { Op } from 'sequelize';
+import { PrismaClient } from '@prisma/client';
+
+const router = express.Router();
+
+// Route for user registration
+router.post('/users/signup', async (req, res) => {
+
+  const { name, password, email, userRole } = req.body;
+
+
+  try {
+    // Check if email already exists
+    const existingUser = await User.findOne({
+      where: {
+        [Op.or]: [{ email }]
+      }
+    });
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Email already exists' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    // Create a new user
+    const newUser = await User.create({
+      name,
+      email,
+      password: hashedPassword,
+      userRole
+    });
+
+    // Set the user in the session
```

Comment:
I like the use of comments across the files, please keep this up so the code is easier to understand